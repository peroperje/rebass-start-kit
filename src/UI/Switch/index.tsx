import * as rebass from "@rebass/forms";

export type SwitchPops = rebass.SwitchProps;

const Switch = rebass.Switch;
export default Switch;
