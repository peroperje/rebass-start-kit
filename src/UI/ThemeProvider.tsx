import React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import theme from "./theme";

export interface Props {
  children: React.ReactNode;
}

const ThemeProvider: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
