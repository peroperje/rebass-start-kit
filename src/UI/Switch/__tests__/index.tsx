import React from "react";
import render from "react-test-renderer";

import Switch from "../index";
import ThemeProvider from "../../ThemeProvider";

describe("Switch", () => {
  it("Default snapshot", () => {
    const tree = render
      .create(
        <ThemeProvider>
          <Switch>Test</Switch>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
