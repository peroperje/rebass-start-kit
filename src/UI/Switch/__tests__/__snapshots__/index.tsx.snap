// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Default snapshot 1`] = `
.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin: 0;
  padding: 0;
  width: 40px;
  height: 24px;
  color: #07c;
  background-color: transparent;
  border: 1px solid;
  border-color: #07c;
  border-radius: 9999px;
}

.emotion-1[aria-checked=true] {
  background-color: #07c;
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: -1px;
  margin-left: -1px;
  width: 24px;
  height: 24px;
  border-radius: 9999px;
  border: 1px solid;
  border-color: #07c;
  background-color: #fff;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
}

<button
  className="emotion-1"
  role="switch"
  type="button"
>
  <div
    aria-hidden={true}
    className="emotion-0"
    style={
      Object {
        "transform": "translateX(0)",
      }
    }
  />
</button>
`;
